grakn-engine/src/main/java/ai/grakn/engine/task/postprocessing/PostProcessingTask.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Grakn - A Distributed Semantic Database
+ * Copyright (C) 2016-2018 Grakn Labs Limited
+ *
+ * Grakn is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Grakn is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Grakn. If not, see <http://www.gnu.org/licenses/gpl.txt>.
+ */
+
+package ai.grakn.engine.task.postprocessing;
+
+import ai.grakn.GraknConfigKey;
+import ai.grakn.GraknTx;
+import ai.grakn.GraknTxType;
+import ai.grakn.Keyspace;
+import ai.grakn.concept.ConceptId;
+import ai.grakn.engine.GraknConfig;
+import ai.grakn.engine.factory.EngineGraknTxFactory;
+import ai.grakn.engine.task.BackgroundTask;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Set;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+/**
+ * <p>
+ *     Class which facilitates running {@link PostProcessor} jobs.
+ * </p>
+ *
+ * @author Filipe Peliz Pinto Teixeira
+ */
+public class PostProcessingTask implements BackgroundTask{
+    private static final Logger LOG = LoggerFactory.getLogger(PostProcessingTask.class);
+    private final EngineGraknTxFactory factory;
+    private final PostProcessor postProcessor;
```

Comment:
All accesses to `postProcessor` just call `index()` - maybe we should just pass in the `IndexPostProcessor`.

Also, does that mean this class doesn't handle the update counts? Where does that happen?

Reply:
Good point!